### ISSUE TITLE
Use flask's application context instead of request context with the db.

### ISSUE THREAD

**Issue by zenhack** [CCI-MOC/hil#733]
See also the discussion on #730. This *should* be enough for us to update to Flask-SQLAlchemy 2.2

**Comment by xuhang57** [comment]
PR merged. Close this now. Please reopen if necessary

**Comment by zenhack** [comment]
@xuhang57, is there a specific pr you're referring to? #730 was a workaround, this issue refers to the more permanent solution.

**Comment by xuhang57** [comment]
@zenhack  sorry, I didn't read the entire discussion. I see this links to a PR and that PR is merged. thanks for keeping this open. It would be great if you could add some actionable tasks on this issue.

**Comment by zenhack** [comment]
Quoting Lucas H. Xu (2018-02-07 20:44:16)

>    [1]@zenhack sorry, I didn't read the entire discussion. I see this
>    links to a PR and that PR is merged. thanks for keeping this open. It
>    would be great if you could add some actionable tasks on this issue.

The title seems straightforward enough to me; we just need to go through
and find everywhere we're creating a request context to use the
database, and create an app context instead.

**Comment by xuhang57** [comment]
True. Curious why we want to use application contexts rather than request contexts? Do we have multiple flask apps or we are going to have? Not so familiar with flask but just skimming through its docs.

**Comment by zenhack** [comment]
See the discussion in #730; basicaly, Flask-SQLAlchemy 2.2 changed things so that you need to use an app context instead of a request context. We're pinning the version for now but we probably don't want to be stuck on an old version forever.

**Comment by xuhang57** [comment]
thanks. So this is basically involved with upgrading `Flask` and some necessary changes while doing the upgrade.
